File: android/src/main/kotlin/dev/note11/flutter_naver_map/flutter_naver_map/util/DisplayUtil.kt
package dev.note11.flutter_naver_map.flutter_naver_map.util

import android.content.res.Resources
import androidx.annotation.Px
import kotlin.math.roundToInt

internal object DisplayUtil {
    private val density: Float
        get() {
            val density = Resources.getSystem().displayMetrics.density
            if (density != 0.0F) return density
            else throw Exception("getDisplayDensity Failed")
        }

    @Px
    fun dpToPx(dp: Double): Int {
        return (dp * density).roundToInt()
    }

    @Px
    fun dpToPxF(dp: Double): Float {
        return (dp * density).toFloat()
    }

    fun pxToDp(@Px px: Int): Double {
        return px.toDouble() / density
    }

    fun pxToDp(@Px px: Float): Double {
        return px.toDouble() / density
    }
}

File: ios/Classes/util/DisplayUtil.swift
//
//  DisplayUtil.swift
//  flutter_naver_map
//
//  Created by 김소연 on 5/29/25.
//

internal struct DisplayUtil {
    static private lazy var _density: CGFloat = {
        return UIScreen.main.scale
    }
    
    static var scale: CGFloat {
        get {
            return _density
        }
    }
    
    static func dpToPx(dp: Double) -> Double {
        return dp * scale
    }
}
